modules/videoeditor-ffmpeg-impl/src/main/java/org/opencastproject/videoeditor/impl/VideoEditorServiceImpl.java

```diff
@@ -265,7 +259,12 @@ protected Track processSmil(Job job, Smil smil, String trackParamGroupId) throws
                     index = srcIndex; // default src
                   }
 
-                  videoclips.add(new VideoClip(index, begin / 1000.0, end / 1000.0));
+                  // Sort out ref elements
+                  if (media.getMediaType() == SmilMediaElement.MediaType.REF) {
+                    refElements.add(new VideoClip(index, begin, end));
```

Comment:
Agreed, will rewrite

Reply:
Okay, so i tried to change this to make more similar by having `new VideoClip(index, begin / 1000.0, end / 1000.0));` both times, but that actually screws the calculations over?

I am also not quite sure yet how you arrived at the conclusion that the constructor treats `double` as `ms` and `long` as `s` respectively. The constructor merely casts long to double, doesn't it?

Reply:
Update: Lars was looking at the constructors of a different `VideoClip` class. Also, the code is correct anyway, since the videoclips are later used for ffmpeg commands which expects seconds, while the subtite classes work with milliseconds. Added a comment to reflect that.